raiden/raiden_event_handler.py

```diff
@@ -543,6 +558,29 @@ def handle_contract_send_channelsettle(
             block_identifier='latest',
         )
 
+    def update_pfs(self, raiden: RaidenService, event: Event):
+        channel_state = get_channelstate_by_token_network_and_partner(
+            chain_state=state_from_raiden(raiden),
+            token_network_id=to_canonical_address(
+                event.balance_proof.token_network_identifier,
+            ),
+            partner_address=to_canonical_address(event.recipient),
+        )
+        if channel_state is not None:
```

Comment:
I think this should be an assertion. Because if the channel no longer exists, there should be no reason to have events emitted which are related to the channel in question.